addons/mail/models/res_users.py

```diff
@@ -80,8 +80,9 @@ def create(self, vals_list):
                 raise exceptions.RedirectWarning(msg, action.id, _('Go to the configuration panel'))
 
         users = super(Users, self).create(vals_list)
-        # Auto-subscribe to channels
-        self.env['mail.channel'].search([('group_ids', 'in', users.groups_id.ids)])._subscribe_users()
+        # Auto-subscribe to channels unless skip explicitly requested
+        if not self.env.context.get('mail_channel_user_nosubscribe', False):
```

Comment:
Indeed double negative are sometimes a bit difficult to understand. However I prefer keeping it like that to tell that its purpose is to skip something, and that default behavior is do to it. I renamed it to mail_channel_skip_auto_subscribe to be clearer.

Reply:
I still think it's better no double negative but new name is already better.
I think the default behavior is simply shown through the default value the key takes no matter what. It can never be shown through the name only.